drivers/led/led_gpio.c

```diff
@@ -24,6 +27,13 @@ struct led_dts_cfg {
 struct led_gpio_data {
 	struct device *gpio_dev;
 	const struct led_dts_cfg *cfg;
+#if defined(CONFIG_LED_SOFTWARE_BLINK)
```

Comment:
Maybe it would be interesting to move this software blink support directly into the LED API. This way every LED controllers without hardware blinking capability could use it (and not only led_gpio). If the feature is enabled, then led_blink() could run the hardware method 'api->blink()' if available and fallback on the software implementation else.